src/createServer.js

```diff
@@ -1,11 +1,252 @@
+    res.send(user);
+  });
+
+  app.delete('/users/:id', async (req, res) => {
+    const { id } = req.params;
+
+    const targetUser = users.find((user) => user.id === +id);
+
+    if (!targetUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    const newUsers = users.filter((user) => user.id !== +id);
+
+    users = newUsers;
+    res.sendStatus(204);
+  });
+
+  app.patch('/users/:id', express.json(), async (req, res) => {
+    const { id } = req.params;
+    const { name } = req.body;
+
+    const targetUser = users.find((user) => user.id === +id);
+
+    if (!targetUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    if (typeof name !== 'string') {
+      res.sendStatus(422);
+
+      return;
+    }
+
+    Object.assign(targetUser, { name });
+
+    res.send(targetUser);
+  });
+
+  let expenses = [];
+  // let expenses = [
+  //   {
+  //     id: 1,
+  //     userId: 2,
+  //     spentAt: '2024-11-29T11:54:36.814Z',
+  //     title: 'Notebook',
+  //     amount: 15,
+  //     category: 'Notebooks',
+  //     note: 'Notebook from AH',
+  //   },
+  //   {
+  //     id: 2,
+  //     userId: 2,
+  //     spentAt: '2024-11-29T11:54:36.814Z',
+  //     title: 'Pen',
+  //     amount: 4,
+  //     category: 'Other',
+  //     note: 'Pen from AH',
+  //   },
+  // ];
+
+  app.get('/expenses', express.json(), async (req, res) => {
+    const { userId, from, to, categories: category } = req.query;
+    let filteredExpenses = [...expenses];
+
+    if (userId) {
+      filteredExpenses = filteredExpenses.filter(
+        (expense) => expense.userId === +userId,
+      );
+    }
+
+    if (from && to) {
+      filteredExpenses = filteredExpenses.filter((expense) => {
+        const expenseDate = Date.parse(expense.spentAt);
+        const fromDate = Date.parse(from);
+        const toDate = Date.parse(to);
+
+        return expenseDate >= fromDate && expenseDate <= toDate;
+      });
+    }
+
+    if (category) {
+      filteredExpenses = filteredExpenses.filter(
+        (expense) => expense.category === category,
+      );
+    }
+
+    res.send(filteredExpenses);
+  });
+
+  app.post('/expenses', express.json(), async (req, res) => {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    if (
+      typeof userId !== 'number' ||
+      typeof spentAt !== 'string' ||
+      typeof title !== 'string' ||
+      typeof amount !== 'number' ||
+      typeof category !== 'string' ||
+      typeof note !== 'string'
+    ) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const assignedUser = users.find((user) => user.id === +userId);
+
+    if (!assignedUser) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const newId = Math.max(...expenses.map((item) => item.id), 0) + 1;
+    const newExpense = {
+      id: newId,
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    };
+
+    expenses.push(newExpense);
+
+    res.statusCode = 201;
```

Comment:
Instead of setting `res.statusCode = 201;`, use `res.status(201);` to set the status code. This is the recommended way to set status codes in Express.